src/Annotations/AppKernel.php

```diff
@@ -0,0 +1,292 @@
+            'process_memory_limit' => $config['process_memory_limit'] ?? 256,
+            'aws' => $config['aws'] + [
+                'queue_name' => 'annotations--queue',
+                'queue_message_default_type' => 'profile',
+                'credential_file' => true,
+                'region' => 'us-east-1',
+                'endpoint' => 'http://localhost:4100',
+             ],
+            'hypothesis' => $config['hypothesis'] + [
+                'api_url' => 'https://hypothes.is/api/',
+                'client_id' => '',
+                'secret_key' => '',
+                'authority' => '',
+            ],
+        ]);
+
+        $this->app->register(new ApiProblemProvider());
+        $this->app->register(new PingControllerProvider());
+
+        if ($this->app['debug']) {
+            $this->app->register(new VarDumperServiceProvider());
+            $this->app->register(new HttpFragmentServiceProvider());
+            $this->app->register(new ServiceControllerServiceProvider());
+            $this->app->register(new TwigServiceProvider());
+            $this->app->register(new WebProfilerServiceProvider(), [
+                'profiler.cache_dir' => __DIR__.'/../../var/cache/profiler',
+                'profiler.mount_prefix' => '/_profiler',
+            ]);
+        }
+
+        // General cache.
+        $this->app['cache'] = function () {
+            return new FilesystemCache(__DIR__.'/../../var/cache');
+        };
+
+        $this->app['logger'] = function (Application $app) {
+            $factory = new LoggingFactory($app['logging.path'], 'annotations-api', $app['logging.level']);
+
+            return $factory->logger();
+        };
+
+        $this->app['monitoring'] = function () {
+            return new Monitoring();
+        };
+
+        /*
+         * @internal
+         */
+        $this->app['limit._memory'] = function (Application $app) {
+            return MemoryLimit::mb($app['process_memory_limit']);
+        };
+        /*
+         * @internal
+         */
+        $this->app['limit._signals'] = function () {
+            return SignalsLimit::stopOn(['SIGINT', 'SIGTERM', 'SIGHUP']);
+        };
+
+        $this->app['limit.long_running'] = function (Application $app) {
+            return new LoggingLimit(
+                new CompositeLimit(
+                    $app['limit._memory'],
+                    $app['limit._signals']
+                ),
+                $app['logger']
+            );
+        };
+
+        $this->app['limit.interactive'] = function (Application $app) {
+            return new LoggingLimit(
+                $app['limit._signals'],
+                $app['logger']
+            );
+        };
+
+        $this->app['hypothesis.guzzle'] = function (Application $app) {
+            // Create default HandlerStack
+            $stack = HandlerStack::create();
+            $logger = $app['logger'];
+            if ($app['debug']) {
+                $stack->push(
```

Comment:
If it's in debug mode, I think we can go the whole mile and use https://github.com/guzzle/psr7#function-str to log the whole request
Also, the same is useful for the response (`GuzzleHttp\Middleware::mapResponse()`)

Reply:
Alternatively, we can log in `$notifyingHttpClient->addRequestListener()` 20 lines from here

Reply:
Should we also add it for the calls to the elife api for the profiles requests?

Reply:
At the `debug` level too, could be useful; not a blocker though